core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -27,6 +28,10 @@ import org.w3c.dom.events.Event as JsEvent
 
 private const val GATT_SERVER_DISCONNECTED = "gattserverdisconnected"
 
+public actual fun CoroutineScope.peripheral(
+    advertisement: Advertisement,
+): Peripheral = peripheral(advertisement.bluetoothDevice)
```

Comment:
This is the same implementation as the Android version's Peripheral implementation. I haven't seen much expected/actual implementations. Is it possible for disparate platforms to share implementations?

Reply:
In this particular case. The `bluetoothDevice` property **name** is the same on both platforms, but the **type** is the platform specific type for each platform, so being able to have this as a single function in `common` is not possible unless the `BluetoothDevice` is defined as `expect`/`actual`.

Unfortunately, since the platforms diverge a bit in their representation of the underlying bluetooth device, it's probably better to have this function duplicated than deal with the potential complexity of trying to unify the bluetooth device platform representations.